robot_smach_states/src/robot_smach_states/navigation/navigation.py

```diff
@@ -39,6 +40,45 @@ def execute(self, userdata=None):
         return 'done'
 
 
+class PrepareDrive(smach.State):
+    """
+    Check if the robot is ready to start driving
+    """
+    def __init__(self, robot, timeout=10, reset_head=True):
+        """
+        @param robot: robot interface
+        @param timeout:
+        """
+        smach.State.__init__(self,
+                             outcomes=['done', 'abort', 'preempted'])
+        self.robot = robot
+        self.timeout = timeout
+        self.reset_head = reset_head
+
+    def execute(self, userdata=None):
+        # keep track of the duration
+        tstart = rospy.Time.now()
+
+        # Cancel head goal, we need it for navigation :)
+        if self.reset_head:
+            self.robot.head.close()
+
+        for part in self.robot.parts.itervalues():
+            # ensure all batteries are unplugged
+            if isinstance(part, Battery):
+                while part.is_charging:
+                    if tstart - rospy.Time.now() < rospy.Duration.from_sec(self.timeout):
+                        return 'abort'
+                    if self.preempt_requested():
+                        return 'preempted'
+                    rospy.logwarn("Robot is charging. Please unplug")
+                    self.robot.speech.speak("I am still charging. Please unplug me.")
+                    rospy.sleep(1)
+                rospy.logerr("Robot is charging, aborting navigation")
+                return 'abort'
```

Comment:
This code is reaced when not charging anymore. Am I correct? That is not matching the code.